core/chains/evm/types/types.go

```diff
@@ -53,6 +55,23 @@ type Receipt struct {
 	TransactionIndex  uint            `json:"transactionIndex"`
 }
 
+type EvmFee struct {
+	// legacy fees
+	Legacy *assets.Wei
+
+	// dynamic/EIP1559 fees
+	DynamicFeeCap *assets.Wei
+	DynamicTipCap *assets.Wei
+}
+
+func (fee EvmFee) String() string {
+	return fmt.Sprintf("{Legacy: %s, DynamicFeeCap: %s, DynamicTipCap: %s}", fee.Legacy, fee.DynamicFeeCap, fee.DynamicTipCap)
+}
+
+func (fee EvmFee) ValidDynamic() bool {
```

Comment:
Curious what is this function for and where would this function be used in?

Reply:
I guess my main question is what changed that required this function. Seems like a pretty important function to me `ValidDynamic`

Reply:
This is not required, you're right. I had initially copied this over from evm/gas since importing EvmFee directly would cause a circular import but I've realized that is A) bad reuse of code B) unnecessary since I can use *assets.Wei instead of EvmFee.

Reply:
fixed!